Library/OcAppleKernelLib/PrelinkedContext.c

```diff
@@ -1256,6 +1389,106 @@ PrelinkedInjectKext (
   return EFI_SUCCESS;
 }
 
+EFI_STATUS
+PrelinkedPassKextToLilu (
+  IN OUT PRELINKED_CONTEXT  *Context,
+  IN     UINT8              KCKind,
+  IN     CONST CHAR8        *BundlePath,
+  IN     CONST CHAR8        *InfoPlist,
+  IN     UINT32             InfoPlistSize,
+  IN     CONST CHAR8        *ExecutablePath OPTIONAL,
+  IN     CONST UINT8        *Executable OPTIONAL,
+  IN     UINT32             ExecutableSize OPTIONAL,
+  OUT    CHAR8              BundleVersion[MAX_INFO_BUNDLE_VERSION_KEY_SIZE] OPTIONAL
+  )
+{
+  EFI_STATUS           Status;
+  CHAR16               EFIVarName[64];
+  VOID                 *Buffer;
+  UINT32               EntryLength;
+  LILU_INJECTION_INFO  *InjectionInfo;
+  UINT32               InfoPlistOffset;
+  UINT32               ExecutableOffset;
+  UINTN                LiluInjectionInfoAddr;
+  UINT64               LiluInjectionInfoAddr64;
+  EFI_GUID             Guid = OC_READ_ONLY_VARIABLE_GUID;
+
+  EntryLength      = sizeof (LILU_INJECTION_INFO) + InfoPlistSize + ExecutableSize;
+  InfoPlistOffset  = sizeof (LILU_INJECTION_INFO);
+  ExecutableOffset = InfoPlistOffset + InfoPlistSize;
+
+  // Allocate buffer and setup header
+  LiluInjectionInfoAddr = (UINTN)AllocateRuntimePool (EntryLength);
+
+  Buffer = (VOID *)LiluInjectionInfoAddr;
+  if (!Buffer) {
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  InjectionInfo              = (LILU_INJECTION_INFO *)Buffer;
+  InjectionInfo->Version     = 0;
+  InjectionInfo->EntryLength = EntryLength;
+  if (AsciiStrCpyS (InjectionInfo->BundlePath, sizeof (InjectionInfo->BundlePath), BundlePath) != RETURN_SUCCESS) {
```

Comment:
Function calls in if-directives make their lines quite long and personally I don't like side effects in them.

Reply:
```c
  Status = AsciiStrCpyS (InjectionInfo->BundlePath, sizeof (InjectionInfo->BundlePath), BundlePath);
  if (EFI_ERROR(Status)) {
    FreePool (Buffer);
    return Status;
  }
```
Is this how it should be? @mhaeuser

Reply:
Yes

Reply:
Done